worlds/aquaria/Regions.py

```diff
@@ -1102,59 +1170,199 @@ def __no_progression_hard_or_hidden_location(self) -> None:
+        if options.no_progression_arnassi_ruins:
+            self.__no_progression_arnassi_ruins(options)
+        if (options.objective.value != Objective.option_killing_the_four_gods and
+            options.objective.value != Objective.option_gods_and_creator):
+            if options.no_progression_frozen_veil:
+                self.__no_progression_frozen_veil()
+            if options.no_progression_abyss:
+                self.__no_progression_abyss()
+            if options.no_progression_sunken_city:
+                self.__no_progression_sunken_city()
+            if options.no_progression_body:
+                self.__no_progression_body()
 
     def adjusting_rules(self, options: AquariaOptions) -> None:
         """
         Modify rules for single location or optional rules
         """
-        self.__adjusting_manual_rules()
+        self.__adjusting_manual_rules(options)
         self.__adjusting_soup_rules()
-        if options.light_needed_to_get_to_dark_places:
-            self.__adjusting_light_in_dark_place_rules()
+        self.__no_progression_areas(options)
+        if options.light_needed_to_get_to_dark_places != LightNeededToGetToDarkPlaces.option_off:
+            self.__adjusting_light_in_dark_place_rules(options.light_needed_to_get_to_dark_places)
         if options.bind_song_needed_to_get_under_rock_bulb:
-            self.__adjusting_under_rock_location()
-
-        if options.mini_bosses_to_beat.value > 0:
-            add_rule(self.multiworld.get_entrance(self.get_entrance_name(self.final_boss_loby, self.final_boss),
-                                                  self.player), lambda state: _has_mini_bosses(state, self.player))
-        if options.big_bosses_to_beat.value > 0:
-            add_rule(self.multiworld.get_entrance(self.get_entrance_name(self.final_boss_loby, self.final_boss),
-                                                  self.player), lambda state: _has_big_bosses(state, self.player))
-        if options.objective.value == options.objective.option_obtain_secrets_and_kill_the_creator:
-            add_rule(self.multiworld.get_entrance(self.get_entrance_name(self.final_boss_loby, self.final_boss),
-                                                  self.player), lambda state: _has_secrets(state, self.player))
+            self.__adjusting_under_rock_location(options)
+        if (options.objective.value != Objective.option_killing_the_four_gods and
+            options.objective.value != Objective.option_gods_and_creator):
+            if options.mini_bosses_to_beat.value > 0:
+                add_rule(self.multiworld.get_entrance(self.get_entrance_name(self.final_boss_lobby, self.final_boss),
+                                                      self.player), lambda state: _has_mini_bosses(state, self.player))
+            if options.big_bosses_to_beat.value > 0:
+                add_rule(self.multiworld.get_entrance(self.get_entrance_name(self.final_boss_lobby, self.final_boss),
```

Comment:
Another comparison with these option didn't use `.value`. Either way should work, but maybe it should be made consistent throughout.